packages/sources/coinpaprika-test/src/endpoint/global.ts

```diff
@@ -0,0 +1,139 @@
+import overrides from '../config/overrides.json'
+
+const logger = makeLogger('CoinPaprika Global Batched')
+
+export const inputParameters = {
+  market: {
+    aliases: ['to', 'quote'],
+    description: 'The symbol of the currency to convert to',
+    required: true,
+    type: 'string',
+  },
+  resultPath: {
+    description: 'The path to the result within the asset quote in the provider response',
+    required: false,
+    type: 'string',
+    options: ['globalmarketcap', 'dominance'],
+  },
+} satisfies InputParameters
+
+export interface GlobalRequestParams {
+  market: string
+  resultPath: '_dominance_percentage' | 'market_cap_'
+}
+
+export interface GlobalResponseBody {
+  market_cap_usd: number
+  volume_24h_usd: number
+  bitcoin_dominance_percentage: number
+  cryptocurrencies_number: number
+  market_cap_ath_value: number
+  market_cap_ath_date: string
+  volume_24h_ath_value: number
+  volume_24h_ath_date: string
+  market_cap_change_24h: number
+  volume_24h_change_24h: number
+  last_updated: number
+}
+
+export type GlobalEndpointTypes = {
+  Request: {
+    Params: GlobalRequestParams
+  }
+  Response: SingleNumberResultResponse
+  Settings: typeof config.settings
+  Provider: {
+    RequestBody: never
+    ResponseBody: GlobalResponseBody
+  }
+}
+
+// Maps the input parameter value with the value that will be set in the requestContext.data object
+const resultPathMap = {
+  globalmarketcap: 'market_cap_',
+  dominance: '_dominance_percentage',
+} as const
+
+const marketMap: { [key: string]: string } = {
+  BTC: 'bitcoin',
+}
+
+const httpTransport = new HttpTransport<GlobalEndpointTypes>({
+  prepareRequests: (params, settings) => {
+    return {
+      params,
+      request: {
+        baseURL: getApiEndpoint(settings),
+        url: '/v1/global',
+        method: 'GET',
+        headers: getApiHeaders(settings),
+      },
+    }
+  },
+  parseResponse: (params, res) => {
+    const data = res.data
+
+    return params.map((p) => {
+      if (!data) {
+        logger.warn(`The data provider did not send any value for this request`)
+        return {
+          params: p,
+          response: {
+            errorMessage: `The data provider did not send any value when requesting the global endpoint`,
+            statusCode: 400,
+          },
+        }
+      }
+
+      const propertyPath =
+        p.resultPath === '_dominance_percentage'
+          ? `${marketMap[p.market.toUpperCase()]}${p.resultPath}`
+          : `${p.resultPath}${p.market.toLowerCase()}`
+
+      const rawResult = data[propertyPath as keyof GlobalResponseBody]
+      if (!rawResult) {
+        return {
+          params: p,
+          response: {
+            errorMessage: `A value for "${propertyPath}" was not found in the provider response`,
+            statusCode: 400,
+          },
+        }
+      }
+
+      const result = Number(data[propertyPath as keyof GlobalResponseBody])
```

Comment:
Couldn't we just use `rawResult` here?
```suggestion
      const result = Number(rawResult)
```